### ISSUE TITLE
restricting access to vm requests

### ISSUE THREAD

**Issue by isabel-amon** [hpi-swt2/vm-portal#113]
**Beschreibung**
As a wimi **only I** want to be able to request vms so that I am able to receive a vm that fits my needs.

On the vms/requests page, **I only** want to see the requests I submitted.

I want to be able to access the form that I can request a vm with as well as send the request to the admin (vms/requests/new).
After I submit a request I want to be redirected to /vms/requests/. I do not want to be able to change the status of my request, I just want to see it.

**Akzeptanzkriterien**
- [ ]  A Wimi only sees requests that he submitted on vms/requests
- [ ] Only users with the role of wimi can see the request-page (and all buttons leading to it)
- [ ] Only they can submit requests
- [ ] After a request is submitted, the user is redirected to the requests-page
- [ ] The user cannot change the status of his/her request

**Priorität**

**Aufwandsschätzung**

**Comment by skydivin4ng3l** [comment]
partially blocked by #99

**Comment by skydivin4ng3l** [comment]
association of User/wimi to the request is not jet done, has to be discussed with @hpi-swt2/bd-18 who implements this cause #99 is responsible for additional users but in #5 the wimi is not assigned to the request yet.

**Comment by BjoernDaase** [comment]
We will do this in #148 next sprint.

**Comment by lassesteffen** [comment]
Looks like a pretty straightforward task when the blocker is removed.

**Comment by lassesteffen** [comment]
Tasks:

- [ ] hide all buttons based on role
- [ ] restrict all controller actions based on role
- [ ] redirect to index page after successful vm creation
- [ ] filter vms index by creator == current_user
- [ ] filter requests index by creator == current_user
- [ ] block the change of requests status if creator == current_user

**Comment by lassesteffen** [comment]
Still blocked because we still do not know which user has

@BjoernDaase am I missing something? Is your team adding the creator of a request to the VM or can we implement this, without interfering with your teams work?

**Comment by BjoernDaase** [comment]
@lassesteffen I think it's not yet added but I do not see major complications if you add it. We should not have a problem with that.

**Comment by MrModder** [comment]
@lassesteffen Adding the user that created the request really shouldn't be an issue
We might need it anyway further down the line if we only want to notify associated users about their request changes.

**Comment by lassesteffen** [comment]
Okey I'm on it

**Comment by lassesteffen** [comment]
@isabel-amon  A Wimi only sees requests that he submitted on vms/requests. If a Wimi is part of another VM where he did not create the request for, this VM will not be shown on /vms ?

**Comment by skydivin4ng3l** [comment]
@isabel-amon please check if this works when merged into dev
